.github/workflows/test.yml

```diff
@@ -0,0 +1,89 @@
+name: Terraform Test
+
+on:
+  workflow_call:
+    inputs:
+      account_id:
+        description: Account id
+        required: true
+        type: string
+
+      terraform_version:
+        description: Which version of terraform to use
+        type: string
+        required: true
+
+      modules_root:
+        description: Root path that terraform modules (folders) are in
+        type: string
+        required: true
+
+permissions:
+  contents: read
+
+jobs:
+  generate-matrix:
+    name: Find Modules
+    runs-on: [self-hosted, Linux, AWS]
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: Brightspace/third-party-actions@actions/checkout
+        with:
+          sparse-checkout: ${{inputs.modules_root}}
+
+      - name: Find testable folders
+        id: set-matrix
+        working-directory: ${{inputs.modules_root}}
+        run: |
+          # Find all folders with at least one .tf file
+          tf_dirs=$(find . -type f -name '*.tf' -exec dirname {} \; | sort -u)
+          result=()
+          for dir in $tf_dirs; do
+            # Check for tests folder with at least one *.tftest.hcl file
+            if [ -d "$dir/tests" ] && find "$dir/tests" -type f -name '*.tftest.hcl' | grep -q .; then
+              # Remove leading ./ for cleaner output
+              clean_dir="${dir#./}"
+              result+=("\"$clean_dir\"")
+            fi
+          done
+          IFS=, matrix_json="{\"folder\":[${result[*]}]}"
+          echo "matrix=$matrix_json" >> $GITHUB_OUTPUT
+
+  test:
+    name: ${{matrix.folder}}
+    needs: generate-matrix
+    runs-on: [self-hosted, Linux, AWS]
+    timeout-minutes: 10
+
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJSON(needs.generate-matrix.outputs.matrix) }}
+
+    steps:
+      - name: Setup Terraform
+        uses: Brightspace/third-party-actions@hashicorp/setup-terraform
+        with:
+          terraform_version: ${{inputs.terraform_version}}
+
+      - uses: Brightspace/third-party-actions@actions/checkout
+        with:
+          sparse-checkout: ${{inputs.modules_root}}
+
+      - name: Assume role
+        uses: Brightspace/third-party-actions@aws-actions/configure-aws-credentials
+        with:
+          aws-access-key-id: ${{secrets.AWS_ACCESS_KEY_ID}}
+          aws-secret-access-key: ${{secrets.AWS_SECRET_ACCESS_KEY}}
+          aws-session-token: ${{secrets.AWS_SESSION_TOKEN}}
```

Comment:
I believe these secrets need to be defined at the top like this:
https://github.com/Brightspace/terraform-workflows/blob/602c4e445670452d521dbc6ba9b489afc4f959e7/.github/workflows/workflow.yml#L60-L72

workflow.yml doesn't define these AWS creds secrets because it doesn't actually use them - it uses OIDC (indicated [here](https://github.com/Brightspace/terraform-workflows/blob/602c4e445670452d521dbc6ba9b489afc4f959e7/.github/workflows/workflow.yml#L83)) to assume the hub role instead (which then assumes the other roles). Either approach works.

Reply:
That's not necessary if you use `secrets: inherit` ([see #2 here](https://docs.github.com/en/actions/how-tos/reuse-automations/reuse-workflows#using-inputs-and-secrets-in-a-reusable-workflow)), which I think I'd prefer to not use. I'll try this out.

Reply:
Using https://github.com/Brightspace/cm-protected-b/pull/62 I've verified that this works with `secrets: inherit` and explicitly specifying the secrets.

Reply:
Sure - I can get to it if you don't

Reply:
I created https://desire2learn.atlassian.net/browse/INFR-149 so I don't forget.